connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresIntegrationSuite.scala

```diff
@@ -123,4 +132,49 @@ class PostgresIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCT
       )
     }
   }
+
+  test("SPARK-49162: Push down date_trunc function") {
+    def testDateTruncPushdown(format: String, expectedResult: Set[Row]): Unit = {
+      val df = sql(
+        s"""
+            SELECT DATE_TRUNC('$format', time), COUNT(*)
+            | FROM $catalogName.datetime_table
+            | GROUP BY 1
```

Comment:
Thanks for adding new tests! Can we do tests where DATE_TRUNC is a part of the predicate (where DATE_TRUNC = ....) to see if it works when we push down predicates. I am not sure if we do pushdowns in projection.

Reply:
Added tests for that as well

Reply:
Why introduce aggregate here? we just need test predicates.

Reply:
The original query that was used to point out the function wasn't pushed down was similar to this one so i modeled the test case after it. Should we remove the aggregate test cases?

Reply:
Yes. Let's remove them.

Reply:
Done